### PR TITLE
Added decompositions for focal lengths from F

### PR DESCRIPTION
**Pull request by kocurvik**
Added decompositions as mentioned in #89 

There are two methods:
* `focals_from_fundamental` using [Bougnoux formula](https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=710808&casa_token=SVXYLNsT1g0AAAAA:0Y9kjd_bwK5WicSdBA24gXfgj6tdkb3Jp5g6ogMVzFltczWB3l__wGGn7p-rm_lAmicEf-PaSk5B&tag=1)
*  `focals_from_fundamental_iterative` using [our recent iterative method](https://openaccess.thecvf.com/content/CVPR2024/papers/Kocur_Robust_Self-calibration_of_Focal_Lengths_from_the_Fundamental_Matrix_CVPR_2024_paper.pdf)

Both also have python bindings.

They both return two Camera structs. The iterative method also takes the prior camera info as the Camera struct and also outputs the number of iterations. The baseline method takes as input also the principal points of the two cameras.

I tested them on Phototourism with SP+SG matches:
together_phototourism_f

And the same setup on Aachen Day-Night v1.1:
together_aachen_f

### CODE REVIEW

File: PoseLib/decompositions.h
Comment (vlarsson) [inline PoseLib/decompositions.h:1]: Maybe this should be under /misc/. At some point we should also add in homography factorization as well.

**Review by vlarsson — CHANGES_REQUESTED** [review]
Looks great! Maybe we should move decompositions to /misc/ ?

File: PoseLib/decompositions.h
Comment (kocurvik) [reply PoseLib/decompositions.h:1]: I made the change.

**Comment by kocurvik** [comment]
I moved the files to misc and added a part about decompositions into the README, also bumped the version to 2.0.3.

**Review by vlarsson — APPROVED** [review]
Looks good to me!

File: PoseLib/CMakeLists.txt [hunk 1]
```diff
@@ -34,6 +34,7 @@ set(SOURCES
     misc/essential.cc
     misc/re3q3.cc
     misc/colmap_models.cc
+        misc/decompositions.cc
```
Comment (pablospe) [inline PoseLib/CMakeLists.txt:37]: there are some extra spaces here, right?

Comment (kocurvik) [reply PoseLib/CMakeLists.txt:37]: Ah yes, my IDE always adds those. Forgot to remove them after moving the files to misc/

**Pull request merged**